src/pages/reader.vue

```diff
@@ -159,9 +165,29 @@ export default {
     } finally {
       this.isLoading = false;
     }
+    window.addEventListener('beforeunload', this.handleBeforeUnload);
+    window.addEventListener('unload', this.handleBeforeUnload);
+  },
+  beforeDestroy() {
+    window.removeEventListener('beforeunload', this.handleBeforeUnload);
+    window.removeEventListener('unload', this.handleBeforeUnload);
+  },
+  beforeRouteLeave(to, from, next) {
+    this.trackReaderClose();
+    next();
   },
   methods: {
     ...mapActions(['restoreAuthSession']),
+    handleBeforeUnload() {
+      this.trackReaderClose();
+    },
+    trackReaderClose() {
+      if (this.hasTracked || !this.openTimestamp) return;
+      this.hasTracked = true;
+
+      const duration = formatDuration(Date.now() - this.openTimestamp);
+      logTrackerEvent(this, 'Reader', 'ReaderClose', this.classId, duration);
```

Comment:
4th param of `logTrackerEvent` is `value` which is a number, we can put time in seconds as numeric value